packages/suite/src/hooks/suite/useFirmware.ts

```diff
@@ -48,13 +52,39 @@ export const useFirmware = () => {
         modal.context === MODAL.CONTEXT_DEVICE &&
         modal.windowType === 'ButtonRequest_FirmwareCheck';
 
-    const isCurrentlyBitcoinOnly = hasBitcoinOnlyFirmware(originalDevice);
+    // Device may be wiped during firmware type switch because Universal and Bitcoin-only firmware have different vendor headers,
+    // except T1B1 and T2T1. There may be some false negatives here during custom installation.
+    // TODO: Determine this in Connect.
+    const willDeviceBeWiped = () => {
+        const deviceModelInternal = originalDevice?.features?.internal_model;
+
+        return (
+            shouldSwitchFirmwareType &&
+            deviceModelInternal !== undefined &&
+            ![DeviceModelInternal.T1B1, DeviceModelInternal.T2T1].includes(deviceModelInternal)
+        );
+    };
+
+    const deviceWillBeWiped = willDeviceBeWiped();
 
     const confirmOnDevice =
-        (firmware.uiEvent?.type === UI.FIRMWARE_RECONNECT &&
-            firmware.uiEvent.payload.method !== 'wait') ||
+        // Show the confirmation pill before starting the installation using the "wait" or "manual" method,
+        // after ReconnectDevicePrompt is closed and user selects the option to install firmware while in bootloader.
+        // Also in case the device is PIN-locked at the start of the process.
         (firmware.uiEvent?.type === DEVICE.BUTTON &&
-            firmware.uiEvent.payload.code === 'ButtonRequest_FirmwareUpdate');
+            firmware.uiEvent.payload.code !== undefined &&
+            ['ButtonRequest_FirmwareUpdate', 'ButtonRequest_PinEntry'].includes(
+                firmware.uiEvent.payload.code,
+            )) ||
+        // Show the confirmation pill right after ReconnectDevicePrompt is closed while using the "wait" or "manual" method,
+        // before user selects the option to install firmware while in bootloader
+        // When a PIN-protected device reconnects to normal mode after installation, PIN is requested and the pill is shown.
+        // There is a false positive in case such device is wiped (including PIN) during custom installation.
+        (firmware.uiEvent?.type === UI.FIRMWARE_RECONNECT &&
+            (firmware.uiEvent.payload.target === 'bootloader' ||
+                (firmware.uiEvent.payload.target === 'normal' &&
```

Comment:
I don't know if the previous `firmware.uiEvent.payload.method !== 'wait'` is equivalent to `firmware.uiEvent.payload.target === 'bootloader' || firmware.uiEvent.payload.target === 'normal'`
If you are sure about it, then fine, asking just in case, so nothing is missed.

Reply:
I think the previous condition was either wrong or it worked coincidentally. The new condition is more descriptive and I tried to help it with the comment. But it is still quite messy :(